support-frontend/assets/components/subscriptionCheckouts/stripeForm/stripeForm.jsx

```diff
@@ -203,27 +195,31 @@ const StripeForm = (props: StripeFormPropTypes) => {
   };
 
   const handleCardErrors = () => {
-    // eslint-disable-next-line array-callback-return
-    ['cardNumber', 'cardExpiry', 'cardCvc'].map((field) => {
-      if (cardFieldsData[field].empty === true) {
-        setCardFieldsData(prevData => ({
-          ...prevData,
-          [field]: {
-            ...prevData[field],
-            error: cardFieldsData[field].errorEmpty,
-          },
-        }));
-      } else if (!cardFieldsData[field].complete) {
-        setCardFieldsData(prevData => ({
-          ...prevData,
-          [field]: {
-            ...prevData[field],
-            error: prevData[field].errorIncomplete,
-          },
-        }));
-      }
-      setCardErrors(getAllCardErrors());
-    });
+    const newCardFieldsErrors = [];
+    const newCardFieldsData = Object.entries(cardFieldsData)
+      .reduce((newData, [cardFieldName, existingFieldData]) => {
+        if (existingFieldData.empty) {
+          const error = existingFieldData.errorEmpty;
+          // eslint-disable-next-line no-param-reassign
+          newData[cardFieldName] = {
```

Comment:
Could we do this inside the `if` block to prevent us ignoring `no-param-reassign` rule?

```js
const error = existingFieldData.errorEmpty;

newCardFieldsErrors.push({ field: [cardFieldName], message: error });

return {
 ...newData,
 [cardFieldName]: {
   ...existingFieldData,
   error
 }
};
```

Reply:
I generally try to avoid this kind of thing because it massively increases the time complexity of the reduce- the spread operator performs an internal iteration so you're essentially iterating over your (increasingly large) object again on each iteration of the loop, giving you an O(n<sup>2</sup>) operation for no actual gains in terms of immutability, since you create a brand new object as your starting accumulator value. But in this case the object is so small it's not going to have any real impact, so happy to switch over.